### ISSUE TITLE
JIT 6502: Elite issue with the Pi 4

### ISSUE THREAD

**Issue by hoglet67** [hoglet67/PiTubeDirect#151]
On the Pi 4, the text on the Elite startup screen is messed up.
![capture11](https://user-images.githubusercontent.com/1951463/148428418-937c6ec7-62d1-42ab-a94a-570343839d80.png)

It should look like this:
![capture8](https://user-images.githubusercontent.com/1951463/148428403-7e2e257f-f010-46d9-8405-11092cf92f75.png)

Pi 3 seems fine.

Introduced between alpha4 and beta1, still present in hognose-dev.

Bisecting implicates this commit: ff0b6fde - which changes quite a few things.

You need to manually apply f98b642 to the above commit, as it also contains a cache bug.

This commit also seems to trigger a failure of the Dorman tests which I think is a seperate issue; see #152

**Comment by hoglet67** [comment]
This is the Dormann failure (report_error at the end)
```
                        ; OR
6b97 : a203                     ldx #3          ;immediate - self modifying code
6b99 : b560             tora    lda zpOR,x
6b9b : 8d1032                   sta ex_orai+1   ;set ORA # operand
                                set_ax  absORa,0
                       >            load_flag 0
6b9e : a900            >            lda #0             ;allow test to change I-flag (no mask)
                       >
6ba0 : 48              >            pha         ;use stack to load status
6ba1 : bd6732          >            lda absORa,x    ;precharge accu
6ba4 : 28              >            plp
                        
6ba5 : 200f32                   jsr ex_orai     ;execute ORA # in RAM
                                tst_ax  absrlo,absflo,0
6ba8 : 08              >            php         ;save flags
6ba9 : dd7332          >            cmp absrlo,x    ;test result
                       >            trap_ne
6bac : f003            >        beq skip3199
                       >        trap            ;failed not equal (non zero)
6bae : 208574          >        jsr report_error
```

**Comment by hoglet67** [comment]
Fixed by fix for #152
